scripts/gen_release_notes/git_parser/git_parser.py

```diff
@@ -0,0 +1,270 @@
+"""
+This module contains all of the methods used to interact with git
+"""
+
+import os
+
+import generate
+import requests
+from helper import helper
+
+# Secrets
+# These are set via ENV variables
+pat = os.getenv("GITHUB_TOKEN", "")
+
+pr_numb_regex = r"^#[0-9]+$"
+merge_commit_regex = r"^Merge"
+cherry_commit_regex = r"\(cherry picked from commit \w+\)"
+
+loading_icons = ["|", "/", "-", "\\"]
+
+
+def get_all_changed_files(repo_owner, repo_name, pr_number):
+    """
+    This function grabs all of the changed files that are in a given PR, given the following:
+    - repo owner
+    - repo name
+    - pr number
+
+    Returns a list of all changed files. If it cannot find any,
+    """
+
+    payload = {"Authorization": "token {secret}".format(secret=pat)}
+    iterator = 1
+    result = list()
+
+    while True:
+
+        try:
+            res = requests.get(
+                "https://api.github.com/repos/{owner}/{repo}/pulls/{pr}/files?per_page=100&page={counter}".format(
+                    owner=repo_owner, repo=repo_name, pr=pr_number, counter=iterator
+                ),
+                headers=payload,
+                timeout=2.0,
+            )
+        except requests.exceptions.RequestException:
+            print("Error getting files from PR. Did you pass in the right values? Or are you authenticated?")
+            return []
+
+        result = result + res.json()
```

Comment:
Not urgent but FYI - this can be `result.extend(res.json())`.

Reply:
Ooo nice, thanks for the tip!